### ISSUE TITLE
Problem in finding processing profiles with the same name as their containing directory

### ISSUE THREAD

**Issue by agriggio** [Beep6581/RawTherapee#3727]
It seems there is an issue in the logic that finds processing profiles, when
the profile name (i.e. the file name without extension) is the same as the
name of the parent directory.

To reproduce: set e.g. the bundled `Pop/Pop` profile as default for RAW images
(preferences -> image processing), and restart RT. At startup, the following
appears:

![rt-error](https://cloud.githubusercontent.com/assets/11982086/23585979/275698a0-018c-11e7-856f-5846d81eaf69.png)

Tested on GIT version: `ac882d2ff22586a96fc97a2fb95679901880503e` (`dev` branch)

```
Compiler: cc 5.4.0
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V3.18.0
Build type: Release
Build flags: -fdiagnostics-color=always -std=gnu++11 -march=native -Werror=unused-label -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG -ffast-math
Link flags:  -march=native
OpenMP support: ON
MMAP support: ON
```

**Comment by gaaned92** [comment]
Cannot reproduce on W10 using pop/pop.pp3 as default for raw image.
```
Version: 5.0-r1-gtk3-76-g3684ac40
Branch: dev
Commit: 3684ac40
Commit date: 2017-03-04
Compiler: gcc 6.3.0
Processor: generic x86
System: Windows
Bit depth: 64 bits
Gtkmm: V3.22.0
Build type: release
Build flags:  -m64 -mwin32 -mthreads  -Werror=unused-label -Werror=unknown-pragmas -Wno-aggressive-loop-optimizations -std=c++11 -mtune=generic -Werror=unused-label -fopenmp -Werror=unknown-pragmas -mwindows -DNDEBUG -msse2 -O3
Link flags: -m64 -mthreads  -static-libgcc   -mtune=generic -mwindows -s -O3
OpenMP support: ON
MMAP support: ON
```

**Comment by TooWaBoo** [comment]
Confirmed, Windows 10 x64

![unbenannt](https://cloud.githubusercontent.com/assets/13800920/23586991/093d16a8-01a2-11e7-97a3-d17ee5f73371.PNG)

```
Version: 5.0-r1-gtk3-82-ga42f5f4a
Branch: dev
Commit: a42f5f4a
Commit date: 2017-03-04
Compiler: gcc 6.3.0
Processor: undefined
System: Windows
Bit depth: 64 bits
Gtkmm: V3.22.0
Build type: release
Build flags: -mwin32 -m64 -mthreads -msse2 -std=c++11 -march=native -Werror=unused-label -fopenmp -Werror=unknown-pragmas -mwindows -Wno-aggressive-loop-optimizations -DNDEBUG -O3
Link flags: -m64 -mthreads -static-libgcc -march=native -mwindows -s -O3
OpenMP support: ON
MMAP support: ON
```

**Comment by gaaned92** [comment]
@agriggio @TooWaBoo 
The problem is that the first name in the list ist not a profile name but the name of the list ( name of the subdir containing the profiles). It should not be selectable in the preferences as it doesn't identify a profile. Could you file a bug report?
But if you have in the list (pop for instance) a pop.pp3 file, its ok.

**Comment by TooWaBoo** [comment]
> ... but the name of the list ( name of the subdir containing the profiles)...

I suggest to remove this **non file entry** from the list. It's only confusing.

**Comment by heckflosse** [comment]
@gaaned92 good find :+1: Imho it should not be selectable in profile selector in general, not only in preferences.

**Comment by heckflosse** [comment]
I found the [culprit](https://github.com/Beep6581/RawTherapee/blob/dev/rtgui/profilestore.cc#L574)

**Comment by heckflosse** [comment]
Suggested workaround...
```
diff --git a/rtgui/profilestore.cc b/rtgui/profilestore.cc
index 7eaf50b..364ae5c 100644
--- a/rtgui/profilestore.cc
+++ b/rtgui/profilestore.cc
@@ -577,12 +577,12 @@ void ProfileStoreComboBox::refreshProfileList_ (Gtk::TreeModel::Row *parentRow,
                     // creating and assigning the custom Label object
                     newSubMenu[methodColumns.label] = entry->label;
                     newSubMenu[methodColumns.profileStoreEntry] = entry;
-
+#if GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION == 18
                     // HACK: Workaround for bug in Gtk+ 3.18...
                     Gtk::TreeModel::Row menuHeader = *(refTreeModel->append(newSubMenu->children()));
-                    menuHeader[methodColumns.label] = entry->label;
+                    menuHeader[methodColumns.label] = "-";
                     menuHeader[methodColumns.profileStoreEntry] = entry;
-
+#endif
                     refreshProfileList_ (&newSubMenu, entry->folderId, false, entryList);
                 } else {
                     refreshProfileList_ (parentRow, entry->folderId, true, entryList);

```
Can someone on > gkt3.18 please test.

**Comment by heckflosse** [comment]
I forgot to mention that the patch from last post was made on `dynamicprofile` branch.

**Comment by TooWaBoo** [comment]
@heckflosse 

What kind of bug is it in Gtk3.18? Maybe it's obsolete in Gtk3.18 too.

**Comment by Beep6581** [comment]
As I remember it, the first entry in all context menu sub-folders was added there a year or two ago to work around some GTK+ bug. If it stays, it would be great if it was disabled/not-selectable/grayed-out.

**Comment by heckflosse** [comment]
@TooWaBoo It's a crash bug when selection profiles. I tested on Gtk3.18 and it crashes without that hack

**Comment by gaaned92** [comment]
@heckflosse patch applied on last dev. The subfolder name doesn't appear in preferences nor in profiles selection.
 OK on gtk3.22

**Comment by heckflosse** [comment]
@gaaned92 Thanks for testing :+1: 
I will try to find a solution to disable the rows for gtk 3.18

**Comment by Beep6581** [comment]
Patch makes sub-folder headers disappear correctly using GTK+ 3.22.5.

**Comment by Beep6581** [comment]
There has always been a small usability issue with the Film Simulation combobox:

1. Set up RT to use the RT Film Simulation Collection http://rawpedia.rawtherapee.com/Film_Simulation
2. Open a photo, enable the Film Simulation tool and select Color > Agfa > Agfa Precisa 100
3. Focus remains on the film simulation combobox. Click the "up" arrow. The combobox now shows "Agfa", which is the title of the group, but the preview has not changed, the preview still shows the effects of Agfa Precisa 100. If you disable Film Simulation, the preview updates. If you re-enable Film Simulation, the preview stays the same because "Agfa" is an invalid choice.

I find it useful that the combo shows "Agfa", because then I know when it has advanced to the next folder, but the preview should refresh to show no effect, as if Film Simulation was disabled. Either that, or clicking "up" should not show the sub-folder names.

In summary, I think the preview should update as if the tool was disabled when the tool uses an invalid choice. Whether the tool should show invalid choices (such as "Agfa", the name of the sub-group), I could debate both ways, so I'll ignore that.

**Comment by heckflosse** [comment]
There's something strange: Using gtk3.18 without the [hack](https://github.com/Beep6581/RawTherapee/blob/dev/rtgui/profilestore.cc#L574) I get an immediate crash when I click on profile selector.
But for HaldCLUT selector which is basically the same but filled by [different code](https://github.com/Beep6581/RawTherapee/blob/dev/rtgui/filmsimulation.cc#L218) without the hack I got no crash.

I would be glad if someone with better skills in that gtk stuff could have a look at that issue.

**Comment by heckflosse** [comment]
@agriggio Can you include my patch from above into your branch? Though it doesn't solve the issue with profile selection completely for gtk 3.18, it solves it for > 3.18 and makes it less confusing for = 3.18

Ingo

**Comment by agriggio** [comment]
sure! I'll integrate later tonight

**Comment by heckflosse** [comment]
@agriggio Great, thank you :+1:

**Comment by heckflosse** [comment]
@Beep6581 Please create a new Issue for the HalfClut selection `bug`!

**Comment by Beep6581** [comment]
@heckflosse sure, issue #3731 opened.
